contracts/stake/src/state.rs

```diff
@@ -183,20 +181,19 @@ impl StakeState {
         // make call to transfer contract to withdraw a note from this contract
         // containing the value of the stake
         let transfer_module = rusk_abi::transfer_module();
-        let _: bool = self
-            .transact(
-                transfer_module,
-                "wfct",
-                &Wfct {
-                    value,
-                    note: unstake.note,
-                    proof: unstake.proof,
-                },
-            )
-            .expect("Withdrawing note from contract should be successful");
+        let _: bool = rusk_abi::call(
+            transfer_module,
+            "wfct",
+            &Wfct {
+                value,
+                note: unstake.note,
+                proof: unstake.proof,
+            },
+        )
+        .expect("Withdrawing note from contract should be successful");
     }
 
-    pub fn withdraw(self: &mut State<Self>, withdraw: Withdraw) {
+    pub fn withdraw(&mut self, withdraw: Withdraw) {
         if rusk_abi::caller() != rusk_abi::transfer_module() {
```

Comment:
Out of scope:
Should this check verify also that the caller is the "root" caller?